Sources/SignalRClient/DefaultHttpClient.swift

```diff
@@ -8,16 +8,20 @@
 
 import Foundation
 
-class DefaultHttpClient: HttpClientProtocol {
+class DefaultHttpClient: NSObject, HttpClientProtocol {
     private let options: HttpConnectionOptions
-    private let session: URLSession
+    private var session: URLSession!
 
     public init(options: HttpConnectionOptions) {
         self.options = options
-        
         let sessionConfig = URLSessionConfiguration.default
         sessionConfig.timeoutIntervalForRequest = options.requestTimeout
-        self.session = URLSession(configuration: sessionConfig)
+        super.init()
+        self.session = URLSession(
+            configuration: sessionConfig,
+            delegate: self,
+            delegateQueue: nil
+        )
```

Comment:
Wouldn't this create a retain cycle? DefaultHttpClient points to URLSession which points to the same instance of DefaultHttpClient via delegate?